src/main/java/com/airbnb/airpal/modules/AirpalModule.java

```diff
@@ -270,7 +271,8 @@ public AWSCredentials provideAWSCredentials()
     public AmazonS3 provideAmazonS3Client(AWSCredentials awsCredentials)
     {
         if (awsCredentials == null) {
-            return new AmazonS3Client();
+            InstanceProfileCredentialsProvider iamCredentials = new InstanceProfileCredentialsProvider(true);
```

Comment:
From my past experience, new InstanceProfileCredentialsProvider(true) will launch a non-daemon background thread to refresh the credential. It caused my java process never quits cleanly. 

Just heads up, if you test it, the process can exit cleanly, just go ahead. Otherwise, using the default constructor is good.